test/shopping-list.test.js

```diff
@@ -0,0 +1,162 @@
+    })
+  })
+  describe("has a method named addItem", function () {
+    let shoppingList = new List();
+    it("should be a function", function () {
+      expect(shoppingList.addItem).to.be.a("function");
+    })
+    it("should add an Item object to the items array", function () {
+      let steak = new Item('steak', 't-bone steak');
+      shoppingList.addItem(steak);
+      expect(shoppingList.items[0]).to.deep.equal(steak);
+    })
+    it("should return an error with anything other than an Item object", function () {
+      expect(shoppingList.addItem.bind(shoppingList, 'cookies')).to.throw();
+    })
+  })
+  describe("has a method named removeItem", function () {
+    let shoppingList = new List();
+    it("should be a function", function () {
+      expect(shoppingList.removeItem).to.be.a("function");
+    })
+    it("should remove an Item object from the items array", function () {
+      let steak = new Item('steak', 't-bone steak');
+      let cookies = new Item('cookies', 'chocolate chip');
+      shoppingList.items.length = 0;
+      shoppingList.addItem(steak);
+      shoppingList.addItem(cookies);
+      shoppingList.removeItem(steak);
+      expect(shoppingList.items[1]).to.equal(undefined);
+    })
+    it("should return an error if item does not exist in the list", function () {
+      let steak = new Item('steak', 't-bone steak');
+      let cookies = new Item('cookies', 'chocolate chip');
+      shoppingList.items.length = 0;
+      shoppingList.addItem(cookies);
+      expect(shoppingList.removeItem.bind(shoppingList, steak)).to.throw();
+    })
+    it("should remove the last item in the list if there is no input", function () {
+      let steak = new Item('steak', 't-bone steak');
+      let cookies = new Item('cookies', 'chocolate chip');
+      shoppingList.items.length = 0;
+      shoppingList.addItem(cookies);
+      shoppingList.addItem(steak);
+      shoppingList.removeItem();
+      expect(shoppingList.items[1]).to.equal(undefined);
+    })
+    it("should return an error with anything other than an Item object", function () {
+      expect(shoppingList.removeItem.bind(shoppingList, 'cookies')).to.throw();
+    })
+  })
+
+  describe("has a method named render", function () {
+    let shoppingList = new List();
+    let steak = new Item('steak', 't-bone steak');
+    let cookies = new Item('cookies', 'chocolate chip');
+    shoppingList.addItem(cookies);
+    shoppingList.addItem(steak);
+    it("should be a function", function () {
+      expect(shoppingList.render).to.be.a("function");
+    })
+
+    it("should return a string", function () {
+      shoppingList.render();
+      expect(shoppingList.render()).to.be.a("string");
+    })
+
+    it("should start with an <ul> tag", function () {
+      expect(shoppingList.render().slice(0, 4)).to.equal('<ul>');
+
+    })
+    it("should end with an </ul> tag", function () {
+      expect(shoppingList.render().slice(-5)).to.equal('</ul>');
+    })
```

Comment:
Same as with Shopping List Item, the render should be tested to see that the contents of the string match what you would expect.